### ISSUE TITLE
[New Feature]: Provide metadata for installed additional components

### ISSUE THREAD

**Issue by hansmbakker** [microsoft/winget-pkgs#68136]
### Description of the new feature/enhancement

Some installers install additional components that should be maintained updated (and uninstalled) separately afterwards.

In https://github.com/microsoft/winget-pkgs/pull/68036#issuecomment-1200440331 you can see a screenshot where multiple subcomponents can be seen.

Issues now:
- Uninstalling the "umbrella" package will not remove the installed components
- the proposed `Package Identifier` is incorrect e.g. for Torpedo Remote. It should be `TwoNotes.TorpedoRemote` instead of `Remote5`
- I wanted to create a separate manifest for Torpedo Remote because it is updated separately from the driver umbrella package (via https://www.two-notes.com/en/torpedo-series/torpedo-remote/)

**Comment by Trenly** [comment]
> the proposed Package Identifier is incorrect e.g. for Torpedo Remote. It should be TwoNotes.TorpedoRemote instead of Remote5

This is incorrect. The script Levvie has uses the ProductCode when printing the uninstall command. The Package Identifier does not come into play at all

> Uninstalling the "umbrella" package will not remove the installed components

This is an issue you will need to raise with the publisher of the software. Publishers control whoch conponents and sub-components get uninstalled when they uninstall their application. There is nothing we can do about this from a packages perspective

> I wanted to create a separate manifest for Torpedo Remote because it is updated separately from the driver umbrella package

This is a good idea. If it can be installed separately, it should have its own manifest

> Is there a way to tell winget that it should recognize these as being described in other manifests?

Creating the manifest for the application will suffice for winget to match the ARP Entry to the application and determine an upgrade is available.

**Comment by hansmbakker** [comment]
> Creating the manifest for the application will suffice for winget to match the ARP Entry to the application and determine an upgrade is available.

I created a manifest for Torpedo Remote at https://github.com/microsoft/winget-pkgs/pull/68221 and updated it using the manifest.

It installed correctly but `winget list` still shows the incorrect Package Id (it lists Package Id as `Remote5`). Maybe that will be fixed when the manifest is merged in the repo and the winget index is updated?

**Comment by Trenly** [comment]
> I found [this documentation](https://docs.microsoft.com/en-us/windows/package-manager/package/manifest?tabs=minschema%2Cversion-example#:~:text=Package%20installers%20in%20MSI%20format%20use%20product%20codes%20to%20uniquely%20identify%20applications.%20The%20product%20code%20for%20a%20given%20version%20of%20a%20package%20should%20be%20included%20in%20the%20manifest%20to%20help%20ensure%20the%20best%20upgrade%20experience.) saying ProductCode is MSI related, but winget recognized this installer as a nullsoft installer?

Product Code in essence, is the registry key that the ARP Entries get written to. Nullsoft, Inno, and other exe installers typically write to the same key every time whereas msi based installers typically have dynamic product codes that change with each version

> Maybe that will be picked up when the manifest is merged in the repo and the winget index is updated?

Exactly correct!
